app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptTextView.java

```diff
@@ -0,0 +1,193 @@
+package com.kabouzeid.gramophone.views;
+
+import android.animation.ObjectAnimator;
+import android.content.Context;
+import android.os.Handler;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.AppCompatTextView;
+import android.text.TextUtils;
+import android.util.AttributeSet;
+import android.util.Log;
+
+/**
+ * Created by lincoln on 7/16/17.
+ */
+
+public class TouchInterceptTextView extends AppCompatTextView {
+    private static final int RETRUNCATE_DELAY = 600;
+
+    private static final String TAG = "E/TouchInterceptFL";
+    private static final String NULL_VIEWS_EXCEPTION_MESSAGE = "Either textView or scrollView is null. Maybe you " +
+            "forgot to set them using setTouchInterceptHorizontalScrollView and setScrollableTextView " +
+            "via XML? Did you set it to something null?";
+    private static final String NULL_LIST_PARENT = "The ListParent, aka the parent ListView or RecyclerView is null." +
+            "It is highly reccomended you set the ListParent either programmatically or via XML" +
+            "if you're TouchInterceptFrameLayout is associated with any type of ListParent. If your" +
+            "TouchInterceptFrameLayout does not interact with any type of ListParent no need to set it" +
+            "and ignore this message.";
+
+    private TouchInterceptFrameLayout touchInterceptFrameLayout;
+    private TouchInterceptHorizontalScrollView scrollView;
+
+    private String song;
+    private String songTruncated;
+
+    private int textBoundsWidth;
+
+    public TouchInterceptTextView(Context context) {
+        super(context);
+        setTag("TITV");
+        setLongClickable(true);
+        setClickable(true);
+        setSingleLine();
+
+    }
+
+    public TouchInterceptTextView(Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        setTag("TITV");
+        setLongClickable(true);
+        setClickable(true);
+        setSingleLine();
+
+    }
+
+    public TouchInterceptTextView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    public TouchInterceptFrameLayout getTouchInterceptFrameLayout() {
+        return (TouchInterceptFrameLayout) getRootView().findViewWithTag("TIFL");
+    }
+
+    public TouchInterceptHorizontalScrollView getTouchInterceptHorizontalScrollView() {
+        return (TouchInterceptHorizontalScrollView) getParent();
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec)
+    {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+        textBoundsWidth = MeasureSpec.getSize(widthMeasureSpec);
+
+        String currentText = getText().toString();
+        Boolean isUntruncatedSong = currentText.endsWith("\uFEFF");
+
+        if(!currentText.endsWith("\u202F") &&
```

Comment:
It's actually not an ellipses, but I should put something there to make it more clear what it is. It's an invisible unicode character I'm using to make what's been truncated.